src/polyfill.ts

```diff
@@ -253,50 +255,29 @@ export class LockManager {
     }
   }
 
+  private _getAbortError(signal: AbortSignal) {
+    return signal.reason || new DOMException("The request was aborted.", "AbortError");
+  }
+
   private _handleSignalExisted(
-    _options: LockOptions,
+    { signal }: LockOptions,
     reject: (reason?: any) => void,
     request: Request
   ) {
-    if (!(_options.signal instanceof AbortSignal)) {
+    if (!(signal instanceof AbortSignal)) {
       reject(
         new TypeError(
           "Failed to execute 'request' on 'LockManager': member signal is not of type AbortSignal."
         )
       );
       return false;
-    } else if (_options.signal.aborted) {
-      reject(
-        new DOMException(
-          "Failed to execute 'request' on 'LockManager': The request was aborted."
-        )
-      );
+    } else if (signal.aborted) {
+      reject(this._getAbortError(signal));
       return false;
     } else {
-      this._signalOnabort(_options.signal, request);
-    }
-    return true;
-  }
-
-  private _handleExceptionWhenStealIsTrue(
-    _options: LockOptions,
-    reject: (reason?: any) => void
-  ) {
-    if (_options.mode !== LOCK_MODE.EXCLUSIVE) {
-      reject(
-        new DOMException(
-          "Failed to execute 'request' on 'LockManager': The 'steal' option may only be used with 'exclusive' locks."
-        )
-      );
-      return false;
-    }
-    if (_options.ifAvailable === true) {
-      reject(
-        new DOMException(
-          "Failed to execute 'request' on 'LockManager': The 'steal' and 'ifAvailable' options cannot be used together."
-        )
-      );
-      return false;
+      const listener = this._signalOnabort(signal, request);
```

Comment:
Opted for returning a listener instead of mutating `request` inside `_signalOnAbort`